pallets/xcmp-queue/src/lib.rs

```diff
@@ -1129,6 +1163,9 @@ impl<T: Config> XcmpMessageSource for Pallet<T> {
 
 		<OutboundXcmpStatus<T>>::put(statuses);
 
+		if !result.is_empty() {
```

Comment:
It should check whether the number of messages in queue is less than the maximum limit.